addons/crm/models/crm_lead.py

```diff
@@ -990,6 +991,12 @@ def toggle_active(self):
             archived.write({'probability': 0, 'automated_probability': 0})
         return res
 
+    @api.model
+    def action_mark_as_lost(self):
```

Comment:
As we already have "action_set_lost", maybe we could name this one "action_set_lost_with_reason", indicating we are going to call the wizard to log the reason ?

Reply:
Can be done also, good reason to introduce a migration script (rename + force upgrade).